app/services/PsaSchemeDashboardService.scala

```diff
@@ -75,15 +80,44 @@ class PsaSchemeDashboardService @Inject()(
     case _ => Future.successful(None)
   }
 
-  def cards(interimDashboard: Boolean, erHtml:Html, srn: String,
-            lock: Option[Lock], list: ListOfSchemes, ua: UserAnswers)
-           (implicit messages: Messages, request: AuthenticatedRequest[AnyContent]): Future[Seq[CardViewModel]] = {
-    val currentScheme = getSchemeDetailsFromListOfSchemes(srn, list)
-    optionLockedSchemeName(lock).map { otherOptionSchemeName =>
-        if(interimDashboard){
-        Seq(manageReportsEventsCard(srn, erHtml))++ Seq(psaCardForInterimDashboard(srn, ua))
+  def cards(
+             interimDashboard: Boolean,
+             erHtml: Html,
+             srn: String,
+             lock: Option[Lock],
+             list: ListOfSchemes,
+             ua: UserAnswers
+           )(implicit messages: Messages, request: AuthenticatedRequest[AnyContent]): Future[Seq[CardViewModel]] = {
+
+    val currentScheme: Option[SchemeDetails] = getSchemeDetailsFromListOfSchemes(srn, list)
+
+    val pstr = currentScheme match {
+      case Some(cs) if cs.referenceNumber.contains(srn) => cs.pstr.getOrElse("")
+      case None => "Pstr Not Found"
+    }
+
+    val seqErOverviewFuture: Future[String] = pensionSchemeReturnConnector.getOverview(
+      pstr, "PSR", minStartDateAsString, maxEndDateAsString
+    ).map {
+      case x if x.size == 1 =>
+        x.head.psrDueDate.map(date => messages("messages__manage_reports_and_returns_psr_due", date.format(formatter)))
+          .getOrElse("")
+      case x if x.size > 1 =>
+        x.head.psrDueDate.map(_ => messages("messages__manage_reports_and_returns_multiple_due"))
+          .getOrElse("")
+      case _ => ""
+    }
+
```

Comment:
Cyclomatic complexity of 11 exceeds max of 10, move logic to another private method and use method call